### ISSUE TITLE
Triggering The Cart Flyout Menu By Clicking "add to cart" On A Single Product Page

### ISSUE THREAD

**Issue by joshcorti11** [elementor/elementor#5921]
## Prerequisites

- [X ] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X ] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] I press the add to cart button on a single product page, and it does not automatically trigger the cart fly-out to open. I choose to use the Elementor cart icon in my header instead of my themes cart icon because most features are more customizable with Elementor, but the automatic cart flyout feature is available with my theme and not Elementor so I was wondering if you guys would be able to add this feature to the Elementor cart widget. I have attached a screenshot of the function I am talking about. The cart menu, as of right now, is only popping out the side of the screen when I physically click on the cart, but I would like this menu to be triggered when I click "add to cart" on a single product page.

**Describe the solution you'd like**
The ability to trigger the "cart fly-out menu" when "add to cart" is clicked on a single product pages.
**Describe alternatives you've considered**
I have considered the themes cart menu and menu cart pro, but they are not as customizable right now.

**Additional context**
![screen shot 2018-10-16 at 10 16 28 am](https://user-images.githubusercontent.com/44182023/47022961-98601500-d12c-11e8-9dea-5707a934f8c9.png)

**Comment by smile00000** [comment]
Agreed, and our subtotal is not working.

**Comment by smile00000** [comment]
I just received word from element support the following in regards to Menu Cart Issues:

"The Menu Cart widget cannot be merged with the sidebar menu. 
The fix for the issue introduced by WC subscriptions code will redirect to the cart page.
The issue with the hamburger menu looks like a Z-index issue and won't be relevant anymore after the release of Elementor Pro 2.4.0. for the 21st of this month."

**Comment by Stilian1** [comment]
+1

**Comment by studiojpeg** [comment]
+1

**Comment by reachdigitalaus** [comment]
Hi anyone worked out a way to get the side cart to be triggered after adding to cart?

**Comment by DavidTheSimon** [comment]
I also would like to know if there is any solution.

**Comment by Zach-Toth** [comment]
I am working on a hack for this. I've almost got it to work except an issue with having to click twice on the overlay to close the menu. Any help getting this to work would be appreciated...

Add this to your child theme's functions.php: 

```
add_action('wp_footer','custom_event_after_add_to_cart_script');
function custom_event_after_add_to_cart_script(){
    if ( is_shop() || is_product_category() || is_product_tag() || is_product() ):
    ?>
         <script type="text/javascript">
             (function($){
             $( document.body ).on( 'added_to_cart', function(){
	         $('.elementor-menu-cart__container').addClass('elementor-menu-cart--shown');
             });
             })(jQuery);
         </script>
     <?php
     endif;
}
```

To fix the double click on overlay to close issue I've tried adding the below to my jQuery function but then it prevents the menu cart button from re-adding the --shown class again until the DOM is refreshed.

```
$( document.body ).on( 'click', function() {
    $( '.elementor-menu-cart__container' ).removeClass( 'elementor-menu-cart--shown' );
}); 
```

Any thoughts on getting this to work properly?

**Comment by Zach-Toth** [comment]
I've found a solution to this. Check out my gist: https://gist.github.com/Zach-Toth/ca5286dfd550c76ba890231e6869dad5

Works great on product pages but still working out a few kinks on archives.

**Comment by nikkonone** [comment]
+1

**Comment by avivu** [comment]
Hello,
This issue has been resolved in Elementor Pro v3.6.6

Feel free to update

Thanks!
